App.js

```diff
@@ -20,6 +21,24 @@ class App extends React.Component {
       </Provider>
     );
   }
+
+  componentDidMount() {
+    BackHandler.addEventListener('hardwareBackPress', this._onBackPress);
```

Comment:
I didn't do a very good job of giving you a snippet that works. Here is a better one. I think we still want to keep lines 28-30 in there.

```
componentDidMount() {
    BackHandler.addEventListener('hardwareBackPress', this._onBackPress);
  }

  componentWillUnmount() {
   
    BackHandler.removeEventListener('hardwareBackPress', this._onBackPress);
  }

  _onBackPress = () => {
    const {dispatch, nav} = this.props;
    // nav.index 0 means you are at the bottom of the navigation stack
    if (nav.index === 0) {
      return false;
    }
    dispatch(NavigationActions.back());
    return true;
  };
```

Reply:
I have updated the code per your suggestion.